source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -1,4 +1,4 @@
-#if defined(__ANDROID__) || (__APPLE__)
```

Comment:
>if [](start = 1, length = 2)

Am I missing where we use the markdown parser on a textblock? seems like it should be somewhere in this file.

Reply:
I think Becky's change would address that.

---
In reply to: [154206301](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154206301) [](ancestors = 154206301)

Reply:
Becky's change addresses usage in the UWP layer. This change should probably include it for all shared model consumers.

---
In reply to: [154207709](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154207709) [](ancestors = 154207709,154206301)

Reply:
I agree, before I make changes here, let me check with Becky how she is using it, so I wouldn't interfere with her work.

---
In reply to: [154208020](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154208020) [](ancestors = 154208020,154207709,154206301)

Reply:
Ricardo and I talked offline.  It does seems a little strange at first to not have it consumed in the text block, but since converting to HTML is more of an implementation detail than it is a functionality we want to expose on the textblock i'm fine with leaving this as is.

---
In reply to: [154208465](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154208465) [](ancestors = 154208465,154208020,154207709,154206301)

Reply:
For what it's worth, my consumption looks like this:

        MarkDownParser markdownParser(HStringToUTF8(textIn));
        auto htmlString = markdownParser.TransformToHtml();

        HString htmlHString;
        UTF8ToHString(htmlString, htmlHString.GetAddressOf());

        HRESULT hr = xmlDocumentIO->LoadXml(htmlHString.Get());

---
In reply to: [154409975](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154409975) [](ancestors = 154409975,154208465,154208020,154207709,154206301)

Reply:
understood & thanks

---
In reply to: [154409975](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154409975) [](ancestors = 154409975,154208465,154208020,154207709,154206301)